### PR TITLE
Add more useful error message for missing Script Pod

### PR DESCRIPTION
**Pull request by flin-8**
[sc-74274]

# Background

Right we now we just tell the user "The Script Pod for script ticket '{scriptTicket}' could not be found, please retry the action" which doesn't help them find the cause.

Now we tell them about evictions and NFS.

CleanShot 2024-04-26 at 15 35 14@2x

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Contracts/ScriptExitCodes.cs [hunk 1]
```diff
@@ -11,5 +11,9 @@ public static class ScriptExitCodes
         public const int TimeoutExitCode = -44;
         public const int UnknownScriptExitCode = -45;
         public const int UnknownResultExitCode = -46;
+
+        //Kubernetes Agent
+        public const int KubernetesScriptPodNotFound = -81;
```
Comment (flin-8) [inline source/Octopus.Tentacle.Contracts/ScriptExitCodes.cs:16]: New exit code for K8s

**Review by scme0 — APPROVED** [review]
Nice 👍

**Review by LukeButters — APPROVED** [review]
Fine from the SaST's concerns which I think is just the ScriptExitCodes file.

**Review by APErebus — APPROVED** [review]
like it!

**Pull request merged**